### PR TITLE
React 18 test run

### PR DESCRIPTION
**Pull request by joecowton1**
## What does this change?

Test run to try and break dcr by using the old versions of Source with the new React 18 render methods.

## Why?

Risk assessment ahead of proposed update.

### CODE REVIEW

**Comment by joecowton1** [comment]
There is an issue using `renderToString` with React 18 where the styles aren't loaded: https://github.com/emotion-js/emotion/issues/2906

Screenshot 2023-02-26 at 09 24 22

File: dotcom-rendering/src/web/components/CalloutEmbed/Form.test.tsx [hunk 1]
```diff
@@ -230,31 +219,30 @@ describe('Callout from', () => {
...
-		const submitButton = queryByText(
-			'Share with the Guardian',
-		) as HTMLButtonElement;
-		fireEvent.click(submitButton);
-
-		expect(mockSubmit.mock.calls.length).toBe(1);
-		// TODO: test mockSubmit internal
-	});
+	// test('should upload the file', () => {
+	// 	const file = new File(['hello'], 'hello.png', { type: 'image/png' });
```
Comment (joecowton1) [inline dotcom-rendering/src/web/components/CalloutEmbed/Form.test.tsx:223]: Something odd happens here where the file is never created instead file = `{}`

File: dotcom-rendering/src/web/components/Dropdown.test.tsx [hunk 2]
```diff
@@ -113,7 +113,8 @@ describe('Dropdown', () => {
 
 		fireEvent.click(getByRole('button'));
 		expect(ulElement).toBeVisible();
-		container.click();
+
+		fireEvent.click(getByRole('button'));
```
Comment (joecowton1) [inline dotcom-rendering/src/web/components/Dropdown.test.tsx:117]: This seems a reasonable fix give `fireEvent.click()` is used elsewhere

**Comment by joecowton1** [comment]
@mxdvl this is just a test but wanted to run this by you - is there an easy alternative to the `renderToString` in the `BlockQuoteComponent`? (see above comment)

We're not using this approach anywhere else that i can see.

File: dotcom-rendering/src/web/components/CalloutEmbed/Form.test.tsx (hunk 1 above)
Comment (mxdvl) [reply dotcom-rendering/src/web/components/CalloutEmbed/Form.test.tsx:223]: Should you use `test.skip`, here?

**Comment by mxdvl** [comment]
> @mxdvl this is just a test but wanted to run this by you - is there an easy alternative to the `renderToString` in the `BlockQuoteComponent`? (see above comment)

I’m not sure, we really want to insert the `QuoteIcon` inline as a sibling of the text nodes, and we achieve this by manipulating strings instead of JSX. I think `jsx` in `RewrappedComponent` can be imported directly from `@emotion/react`, though:

```diff
- import { ClassNames } from '@emotion/react';
- import { jsx as _jsx } from 'react/jsx-runtime';
+ import { ClassNames, jsx } from '@emotion/react';
```

> We're not using this approach anywhere else that i can see.

There are a few instances:
- [`SecureSignup`](https://github.com/guardian/dotcom-rendering/blob/cbe0ef4bdc8e76e6d324dfd7bd02d8850a7ca217/dotcom-rendering/src/web/components/SecureSignup.tsx#L50-L54)
- [`KeyEventsContainer`](https://github.com/guardian/dotcom-rendering/blob/cbe0ef4bdc8e76e6d324dfd7bd02d8850a7ca217/dotcom-rendering/src/web/server/keyEventsToHtml.tsx#L16-L22)
- [`renderToStringWithEmotion`](https://github.com/guardian/dotcom-rendering/blob/cbe0ef4bdc8e76e6d324dfd7bd02d8850a7ca217/dotcom-rendering/src/web/lib/emotion.tsx#L11-L13)

**Comment by joecowton1** [comment]
> > We're not using this approach anywhere else that i can see.
> 
> There are a few instances:
> 
> * [`SecureSignup`](https://github.com/guardian/dotcom-rendering/blob/cbe0ef4bdc8e76e6d324dfd7bd02d8850a7ca217/dotcom-rendering/src/web/components/SecureSignup.tsx#L50-L54)
> * [`KeyEventsContainer`](https://github.com/guardian/dotcom-rendering/blob/cbe0ef4bdc8e76e6d324dfd7bd02d8850a7ca217/dotcom-rendering/src/web/server/keyEventsToHtml.tsx#L16-L22)
> * [`renderToStringWithEmotion`](https://github.com/guardian/dotcom-rendering/blob/cbe0ef4bdc8e76e6d324dfd7bd02d8850a7ca217/dotcom-rendering/src/web/lib/emotion.tsx#L11-L13)

Interesting, I'd meant replacing the existing html rather than using `renderToString` but i wonder if the styles are broken in the above instances too. 

I think really this is a job for the platform teams as part of React 18 adoption so don't want to get too involved, just thinking about how much of a blocker it would be.

**Pull request closed without merging**